source/common/sagemath/library.sage

```diff
@@ -222,14 +222,13 @@ class TBIL:
 
     @staticmethod
     def typeset_angle(theta):
-        if type(theta) is not sage.symbolic.expression.Expression:
-            return f"${latex(theta)}$"
-        elif not theta.is_rational_expression(): 
-            return f"${latex(theta)}$"
-        elif denominator(theta)==1:
-            return f"${latex(theta)}$"
-        else:
-            return f"$\\dfrac{{{latex(numerator(theta))}}}{{{denominator(theta)}}}$"
+        angle_string=f"${latex(theta)}$"
+        if type(theta) is sage.symbolic.expression.Expression and theta.is_rational_expression() and denominator(theta)!=1:
+            if numerator(theta)<0:
+                angle_string = f"$-\\dfrac{{{latex(-1*numerator(theta))}}}{{{denominator(theta)}}}$"
+            else:
+                angle_string = f"$\\dfrac{{{latex(numerator(theta))}}}{{{denominator(theta)}}}$"
+        return angle_string
```

Comment:
Remove some of the code duplication, but also typeset negative angles better (e.g. $-\frac{\pi}{2}$ instead of $\frac{-\pi}{2}$.